lib/ui/window.dart

```diff
@@ -324,25 +358,32 @@ class FlutterView {
   /// then obtain a [Scene] object, which you can display to the user via this
   /// [render] function.
   ///
+  /// If the view is configured with loose [physicalConstraints] (i.e.
+  /// [ViewConstraints.isTight] returns false) a `size` satisfying those
+  /// constraints must be provided. This method does not check that the provided
+  /// `size` actually meets the constraints (this should be done in a heigher
+  /// level), but an illegal `size` may result in undefined rendering behavior.
+  /// If no `size` is provided, [physicalSize] is used instead.
+  ///
   /// See also:
   ///
   /// * [SchedulerBinding], the Flutter framework class which manages the
   ///   scheduling of frames.
   /// * [RendererBinding], the Flutter framework class which manages layout and
   ///   painting.
-  void render(Scene scene) {
+  void render(Scene scene, {Size? size}) {
```

Comment:
For some historic context: I wish we could assert here that the size passed into this method meets the physicalConstraints of the FlutterView. However, doing that is breaking a ton of things. Among other things, tests freely mock out physicalSize/physicalConstraints, which means the size the end up providing here doesn't match the real physicalConstraints. Changing the test framework to not call render in those situations (i.e. outside of onBeginFrame / onDrawFrame, https://github.com/flutter/flutter/issues/137073) breaks golden tests on web because they rely on being able to call render whenever they want.

Long story short: We decided to not assert that the size meets the constraints here in this low-level API and instead left it to a higher level to ensure that something sensible is passed here.

Reply:
@goderbauer should we have an issue so this can be revisited/worked on by our future selves?

Reply:
This is essentially covered by https://github.com/flutter/flutter/issues/137073, I think. I left a comment to consider adding the assert when we fix that.